src/context/AuthContext.tsx

```diff
@@ -16,20 +15,11 @@ export function useAuth() {
 export function AuthProvider({ children }: React.PropsWithChildren<{}>) {
   const [currentUser, setCurrentUser] = useState<User>(null);
 
-  // function signup(email,password){
-  //   return auth.createUserWithEmailAndPassword(email,password)
-  // }
-
-  auth.onAuthStateChanged((user) => {
-    setCurrentUser(user);
-  });
-
   useEffect(() => {
     const unsubcribe = auth.onAuthStateChanged((firebaseUser) => {
```

Comment:
Question again, does returning here change the behaviour? Or could we do the short `auth.onAuthStateChanged(setCurrentUser)`.

Reply:
[Its a callback function](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function). Honestly passing the function as a callback vs the nested anonymous function (I think thats how they call it, I don't know js) is mostly irrelevant here, both work at the end... 

What I really wanted to learn was if assigning and returning `unsubscribe` served a particular purpose or has some behaviour in this case, I really don't know :) and was not trying to nitpick.

Reply:
I think it's fine. I guess I didn't realize `setUser` could be used as a callback as it is a function itself. If we do run into trouble in the future, we at least know this could be a potential issue.

I looked at different guides and saw more often my initial approach. Take a look [here](https://lo-victoria.com/introduction-to-react-context-api-with-firebase-authentication) and another [one](https://medium.com/geekculture/firebase-auth-with-react-and-typescript-abeebcd7940a)

It may also be more readable because according to the doc `onAuthStateChanged` returns `Firebase.unsubscribe`, but one wouldn't know unless you read [docs](https://firebase.google.com/docs/reference/js/v8/firebase.auth.Auth#onauthstatechanged), I just looked it up.

Take a look and then merge :) I think current state is OK , will add another comment